### PR TITLE
Media system / components cleanup

### PR DESCRIPTION
**Pull request by speigg**
## Summary

- Decouples all media components from one-another ( Image / Video / Volumetric )
- Unifies all playlist management and media element data binding in MediaComponent.ts
- Clean up ComponentFunctions.ts
  - deprecates `createMappedComponent`
  -  adds `defineComponent`
- Improved Editor React typings
- Update hookstate
- Improved Editor support for hookstate-driven component data

## References

closes #_insert number here_

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] [ensure all checks pass](https://github.com/XRFoundation/XREngine/wiki/Testing-&-Contributing)
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/engine/src/scene/systems/SceneObjectUpdateSystem.ts [hunk 1]
```diff
@@ -402,15 +393,17 @@ export default async function SceneObjectUpdateSystem(world: World) {
 
   world.scenePrefabRegistry.set(ScenePrefabs.image, [
     ...defaultSpatialComponents,
-    { name: SCENE_COMPONENT_IMAGE, props: SCENE_COMPONENT_IMAGE_DEFAULT_VALUES }
+    {
+      name: SCENE_COMPONENT_IMAGE,
+      props: { source: '__$project$__/XREngine-development-test-suite/assets/sample_etc1s.ktx2' }
```
Comment (HexaField) [inline packages/engine/src/scene/systems/SceneObjectUpdateSystem.ts:398]: this should be default-project, and having the __$project$__ identifier here will not properly deserialize the first time. this mutation happens on the backend.

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/ecs/functions/EntityFunctions.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export const removeEntity = (entity: Entity, immediately = false, world = Engine
     bitECS.removeEntity(world, entity)
   } else {
     removeAllComponents(entity, world)
-    addComponent(entity, EntityRemovedComponent, {})
+    addComponent(entity, EntityRemovedComponent, null)
```
Comment (HexaField) [inline packages/engine/src/ecs/functions/EntityFunctions.ts:18]: what's the reasoning for this? it will cause hasComponent checks to always return false

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/ecs/functions/EntityFunctions.ts (hunk 2 above)
Comment (speigg) [reply packages/engine/src/ecs/functions/EntityFunctions.ts:18]: it won't

Comment (HexaField) [reply packages/engine/src/ecs/functions/EntityFunctions.ts:18]: whats the reason for doing this instead of `true`

Comment (speigg) [reply packages/engine/src/ecs/functions/EntityFunctions.ts:18]: true takes up space, and isn't necessary

**Pull request merged**